stat/sampleuv/weighted.go

```diff
@@ -54,70 +52,89 @@ func NewWeighted(w []float64, src rand.Source) Weighted {
 // already taken.
 func (s Weighted) Len() int { return len(s.weights) }
 
+func (s Weighted) findIndex(r float64) int {
```

Comment:
Conventionally we place caller before callee, but in this case I'd prefer that this block be put back in `Take`; it's not called from more than one location and doesn't significantly complexify the calling function.

So, as
```
	r *= s.heap[0]
	i := 0
	for {
		r -= s.weights[i]
		if r < 0 {
			break // Fall within item i.
		}
...
```
after `r` decl in `Take`.

Reply:
Fixed